archivers/7zip/Portfile

```
@@ -53,10 +53,15 @@ post-patch {
     reinplace "s|LFLAGS_ALL = $\(\LFLAGS_STRIP\)\|LFLAGS_ALL =|" \
     ${worksrcpath}/C/7zip_gcc_c.mak \
     ${worksrcpath}/CPP/7zip/7zip_gcc.mak
+    # do not force -O2 flag during compilation
+    reinplace "s|CFLAGS_BASE = $\(\MY_ARCH_2\)\ -O2|CFLAGS_BASE = $\(\MY_ARCH_2\)\ ${configure.cflags}|" \
```

`M` and ` ` are not a special characters in regular expressions and do not need a backslash before them.

`(` and `)` are special characters in extended regular expressions but you have not used `reinplace`'s `-E` flag to enable extended regular expressions, and they are not special characters in basic regular expressions.

Within quotation marks, backslashes are consumed by Tcl before reaching the regex engine, and none of these characters are special to Tcl, so all of the backslashes are unnecessary.

`$` is a special character to Tcl and to regular expressions so it should be escaped from both engines by using a double backslash in the regular expression part and a single blackslash in the replacement part.

```suggestion
    reinplace "s|CFLAGS_BASE = \\$(MY_ARCH_2) -O2|CFLAGS_BASE = \$(MY_ARCH_2) ${configure.cflags}|" \
```

Please apply these suggestions to the existing `reinplace`s for `LDFLAGS` and `LFLAGS_ALL`.

---

`Error: Failed to patch 7zip: can't read "(MY_ARCH_2)": no such variable`

I don't think this behavior was expected.